File: 05/CSharp/ExpressionApp/Add.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExpressionApp
{
    class Add : BinaryOperator
    {
        public Add(IExpression first, IExpression second)
        {
            fOp = first;
            sOp = second;
        }
        public override double Evaluate(double param)
        {
            return fOp.Evaluate(param) + sOp.Evaluate(param);
        }
    }
}

File: 05/Java/expression/Add.java
package Expression;

public class Add extends BinaryOperator implements Expression {
    public Add(Expression first, Expression second) {
        fOp = first;
        sOp = second;
    }

    public double evaluate(double param) {
        return fOp.evaluate(param) + sOp.evaluate(param);
    }
}